x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/compare_synced_integrations.ts

```diff
@@ -0,0 +1,324 @@
+    throw err;
+  }
+};
+
+export const fetchAndCompareSyncedIntegrations = async (
+  esClient: ElasticsearchClient,
+  savedObjectsClient: SavedObjectsClientContract,
+  index: string,
+  logger: Logger
+): Promise<GetRemoteSyncedIntegrationsStatusResponse> => {
+  try {
+    // find integrations on ccr index
+    const searchRes = await esClient.search<SyncIntegrationsData>({
+      index,
+      sort: [
+        {
+          'integrations.updated_at': {
+            order: 'desc',
+          },
+        },
+      ],
+    });
+    if (searchRes.hits.hits[0]?._source === undefined) {
+      return {
+        integrations: [],
+        error: `No integrations found on ${index}`,
+      };
+    }
+    const ccrIndex = searchRes.hits.hits[0]?._source;
+    const { integrations: ccrIntegrations, custom_assets: ccrCustomAssets } = ccrIndex;
+
+    // find integrations installed on remote
+    const installedIntegrations = await getPackageSavedObjects(savedObjectsClient);
+
+    const installedIntegrationsByName = (installedIntegrations?.saved_objects || []).reduce(
+      (acc, integration) => {
+        if (integration?.id) {
+          acc[integration.id] = integration;
+        }
+        return acc;
+      },
+      {} as Record<string, SavedObjectsFindResult<Installation>>
+    );
+    const customAssetsStatus = await fetchAndCompareCustomAssets(esClient, logger, ccrCustomAssets);
+    const integrationsStatus = compareIntegrations(ccrIntegrations, installedIntegrationsByName);
+    const result = {
+      ...integrationsStatus,
+      ...(customAssetsStatus && { custom_assets: customAssetsStatus }),
+    };
+
+    return result;
+  } catch (error) {
+    logger.error('error', error?.message);
+    return {
+      integrations: [],
+      error: error?.message,
+    };
+  }
+};
+
+const compareIntegrations = (
+  ccrIntegrations: IntegrationsData[],
+  installedIntegrationsByName: Record<string, SavedObjectsFindResult<Installation>>
+): { integrations: RemoteSyncedIntegrationsStatus[] } => {
+  const integrationsStatus: RemoteSyncedIntegrationsStatus[] | undefined = ccrIntegrations?.map(
+    (ccrIntegration) => {
+      const localIntegrationSO = installedIntegrationsByName[ccrIntegration.package_name];
+      if (!localIntegrationSO) {
+        return {
+          ...ccrIntegration,
+          sync_status: 'synchronizing' as SyncStatus.SYNCHRONIZING,
+        };
+      }
+      if (ccrIntegration.package_version !== localIntegrationSO?.attributes.version) {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Found incorrect installed version ${localIntegrationSO?.attributes.version}`,
+        };
+      }
+      if (localIntegrationSO?.attributes.install_status !== 'installed') {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Installation status: ${localIntegrationSO?.attributes.install_status}`,
```

Comment:
can we return the error message if `install_status: install_failed` from `latest_install_failed_attempts` or `latest_executed_state`? https://github.com/elastic/kibana/blob/0c5fa8465905aad665c91c858f38f980b0013eb4/x-pack/platform/plugins/shared/fleet/common/types/models/epm.ts#L674